src/main/java/org/deri/cqels/engine/BDBGraphPatternRouter.java

```diff
@@ -69,6 +78,16 @@ public BDBGraphPatternRouter(ExecContext context, Op op, DatasetGraph ds) {
 		init(ds);
 	}
 	
+    private boolean isLDFServer(String url){
+        return url.indexOf("sparql")+"sparql".length()!=url.length();               
```

Comment:
Unfortunately, this is not a reliable way to check whether the given URL belongs to a SPARQL endpoint, e.g. the SPARQL endpoint provided by Apache Fuseki doesn't have `sparql` word in its URL.

I've sent a mail to LDF mailing list, let's see what do they say.